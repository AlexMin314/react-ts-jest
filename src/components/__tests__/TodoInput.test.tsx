import * as React from 'react';
import {TodoStore} from '../../stores/TodoStore';
import {TodoInput} from '../TodoInput';
import {mount} from 'enzyme';

describe('TodoInput', () => {
    it('should add todo', () => {
        // Given
        const todoStore = new TodoStore();
        const wrapper = mount(<TodoInput todoStore={todoStore}/>);

        // When
        const input = wrapper.find('input').first();
        const button = wrapper.find('button').first();

        input.simulate('change', {target: {value: 'hello world'}});
        button.simulate('click');

        // Then
        expect(todoStore.todos.length).toBe(1);
        expect((input.getDOMNode() as HTMLInputElement).value).toBe('');
        expect(wrapper.find('div')).toMatchSnapshot();
    });
});
