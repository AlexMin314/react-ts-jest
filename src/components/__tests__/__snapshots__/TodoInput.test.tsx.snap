// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoInput should add todo 1`] = `
<div>
  <input
    name="todoInput"
    type="text"
  />
  <button
    onClick={[Function]}
  >
    Add
  </button>
</div>
`;
